core/src/main/scala/org/apache/spark/sql/eventhubs/EventHubsSource.scala

```diff
@@ -160,18 +160,34 @@ private[spark] class EventHubsSource private[eventhubs] (sqlContext: SQLContext,
     val seqNos: Map[NameAndPartition, SequenceNumber] = maxOffsetsPerTrigger match {
       case None =>
         latest
+
       case Some(limit) if currentSeqNos.isEmpty =>
-        rateLimit(limit, initialPartitionSeqNos, latest, earliestSeqNos.get)
+        val startingSeqNos = adjustStartingOffset(initialPartitionSeqNos)
+        rateLimit(limit, startingSeqNos, latest, earliestSeqNos.get)
+
       case Some(limit) =>
-        rateLimit(limit, currentSeqNos.get, latest, earliestSeqNos.get)
+        val startingSeqNos = adjustStartingOffset(currentSeqNos.get)
+        rateLimit(limit, startingSeqNos, latest, earliestSeqNos.get)
     }
 
     currentSeqNos = Some(seqNos)
-    logDebug(s"GetOffset: ${seqNos.toSeq.map(_.toString).sorted}")
+    logInfo(s"GetOffset: ${seqNos.toSeq.map(_.toString).sorted}")
 
     Some(EventHubsSourceOffset(seqNos))
   }
 
+  private def adjustStartingOffset(
+      from: Map[NameAndPartition, SequenceNumber]): Map[NameAndPartition, SequenceNumber] = {
+    from.map {
+      case (nAndP, seqNo) =>
+        if (seqNo < earliestSeqNos.get(nAndP)) {
```

Comment:
In case the starting offset is adjusted, shouldn’t we report data loss or log this change in the starting offset?